### PR TITLE
Removed query-string, use goto svelte for url state changes

### PR DESCRIPTION
**Pull request by Ak4zh**
- Added new environment variable POST_CATEGORIES
- Added post filters using post categories
- Removed the use of `query-string` lib as the same task can be achieved with svelte '$page.url.searchParams`
- Removed all urlState functions and changed to use svelte `goto` instead.

### CODE REVIEW

File: src/routes/blog/+page.svelte [hunk 1]
```diff
@@ -1,87 +1,47 @@
...
-				show: [
-					// essays && 'Essays',
-					// talks && 'Talks',
-					// podcasts && 'Podcasts',
-					// snippets && 'Snippets',
-					// tutorials && 'Tutorials',
-					// notes && 'Notes'
-				].filter(Boolean)
-			});
-		}, 100);
```
Comment (sw-yx) [inline src/routes/blog/+page.svelte:75]: this code is required for URLs to be copy pastable

eg when i search something on my site it shows up in the URL, and can be copy pasted

image


but on this PR it doesnt

image

File: src/lib/siteConfig.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const DEFAULT_OG_IMAGE =
 	'https://user-images.githubusercontent.com/6764957/147861359-3ad9438f-41d1-47c8-aa05-95c7d18497f0.png';
 export const MY_TWITTER_HANDLE = 'swyx';
 export const MY_YOUTUBE = 'https://youtube.com/swyxTV';
+export const POST_CATEGORIES = ['Blog']
```
Comment (sw-yx) [inline src/lib/siteConfig.js:11]:
```suggestion
export const POST_CATEGORIES = ['Blog'] // Other categories you can consider adding: Talks, Tutorials, Snippets, Podcasts, Notes...
```

**Comment by sw-yx** [comment]
appreciate the PR! left some thoughts before merging, curious what you think. let me know if you are interested in being a maintainer as well - i just ask that you send every change in as a PR so that we can discuss first and I can keep track

File: src/routes/blog/+page.svelte (hunk 1 above)
Comment (Ak4zh) [reply src/routes/blog/+page.svelte:75]: [14691a7](https://github.com/sw-yx/swyxkit/pull/97/commits/14691a773585345df798423901576f72ecf016a8)
Fixed the URLs to be updated with search filter.

**Comment by Ak4zh** [comment]
I think URL parameter name should be changed as below for a more clear naming:
- `show` to `categories` or `tags` (I prefer tags as it's shorter)
- `filter` to `search` or `keyword` (I prefer keyword here as search gives a feeling of path rather that a param)

Let me know if that looks ok.

**Comment by sw-yx** [comment]
almost! when you go to here: https://deploy-preview-97--swyxkit.netlify.app/blog?filter=test&show=Blog

the "show" code is dropped somehow

image

**Comment by sw-yx** [comment]
i tried fixing it but couldnt find the source of the bug - i can get back to it in future, thank you for the contribution, it is very valued.

**Comment by Ak4zh** [comment]
> i tried fixing it but couldnt find the source of the bug - i can get back to it in future, thank you for the contribution, it is very valued.

Fixed.
It seems it was happening because both getting the param get value and set were reactive and running at same time.

File: README.md [hunk 3]
```diff
@@ -111,6 +112,7 @@ export const DEFAULT_OG_IMAGE =
 	'https://user-images.githubusercontent.com/6764957/147861359-3ad9438f-41d1-47c8-aa05-95c7d18497f0.png';
 export const MY_TWITTER_HANDLE = 'swyx';
 export const MY_YOUTUBE = 'https://youtube.com/swyxTV';
+export const POST_CATEGORIES = ['Blog']
```
Comment (sw-yx) [inline README.md:115]:
```suggestion
export const POST_CATEGORIES = ['Blog'] // Other categories you can consider adding: Talks, Tutorials, Snippets, Podcasts, Notes...
```

File: README.md [hunk 4]
```diff
@@ -49,6 +49,7 @@ See https://swyxkit.netlify.app/ (see [Deploy Logs](https://app.netlify.com/site
 - **Minor design/UX touches**
   - Top level blog URLs (`/myblog` instead of `/blog/myblog` - [why](https://www.swyx.io/namespacing-sites/))
   - Blog index truncates at 20 posts to make sure to render quickly
+  - Blog search/facets serialize to URLs for easy copy paste ([thanks @Ak4zh](https://github.com/sw-yx/swyxkit/pull/97))
```
Comment (sw-yx) [inline README.md:52]: added attribution to you @Ak4zh  :)

**Review by sw-yx — APPROVED** [review]

**Comment by sw-yx** [comment]
thank you so much!

**Pull request merged**